`--cppCompileToNamespace:foo` fails compilation with `import os`

<!-- Think about the title, twice. -->

<!-- Summarize the problem here, keep it short and simple. -->
If the Nim project has `import os`, compilation with `--cppCompileToNamespace:foo` fails.



### Example
<!-- Paste your example in the code-block below. -->
```nim
# t.nim
import os

echo "a" / "b"
```

Compile above with:

```
nim cpp --cppCompileToNamespace:foo -r t.nim 
```

### Current Output
```
/tmp/kmodi/.cache/nim/t_d/stdlib_os.cpp:41:1: error: ‘NI’ does not name a type
 NI len;
 ^~
/tmp/kmodi/.cache/nim/t_d/stdlib_os.cpp:42:1: error: ‘NI’ does not name a type
 NI reserved;
 ^~
/tmp/kmodi/.cache/nim/t_d/stdlib_os.cpp:45:1: error: ‘NIM_CHAR’ does not name a type; did you mean ‘NIM_IMAN’?
 NIM_CHAR data[SEQ_DECL_SIZE];
 ^~~~~~~~
 NIM_IMAN
/tmp/kmodi/.cache/nim/t_d/stdlib_os.cpp:47:9: error: ‘NU8’ does not name a type; did you mean ‘NULL’?
 typedef NU8 tyEnum_TNimKind_jIBKr1ejBgsfM33Kxw4j7A;
         ^~~
         NULL
/tmp/kmodi/.cache/nim/t_d/stdlib_os.cpp:48:9: error: ‘NU8’ does not name a type; did you mean ‘NULL’?
 typedef NU8 tySet_tyEnum_TNimTypeFlag_v8QUszD1sWlSIWZz7mC4bQ;
         ^~~
         NULL
/tmp/kmodi/.cache/nim/t_d/stdlib_os.cpp:49:71: error: ‘NI’ has not been declared
 typedef N_NIMCALL_PTR(void, tyProc_ojoeKfW4VYIm36I9cpDTQIg) (void* p, NI op);
                                                                       ^~
/tmp/kmodi/.cache/nim/t_d/stdlib_os.cpp:52:1: error: ‘NI’ does not name a type
```


### Expected Output
<!-- What should be the correct output? -->
```
a/b
```


### Additional Information
```
> nim -v
Nim Compiler Version 0.19.9 [Linux: amd64]
Compiled at 2019-05-06
Copyright (c) 2006-2019 by Andreas Rumpf

git hash: 2475d92c36e26e44372175c2cf4c21fbab084619
active boot switches: -d:release
```

--- a/compiler/cgen.nim
+++ b/compiler/cgen.nim
@@ -1695,13 +1695,12 @@ proc genModule(m: BModule, cfile: Cfile): Rope =
   result = getFileHeader(m.config, cfile)
   result.add(genMergeInfo(m))
 
-  if m.config.cppCustomNamespace.len > 0:
-    result.add openNamespaceNim(m.config.cppCustomNamespace)
-
   generateThreadLocalStorage(m)
   generateHeaders(m)
   add(result, genSectionStart(cfsHeaders, m.config))
   add(result, m.s[cfsHeaders])
+  if m.config.cppCustomNamespace.len > 0:
+    result.add openNamespaceNim(m.config.cppCustomNamespace)
   add(result, genSectionEnd(cfsHeaders, m.config))
   add(result, genSectionStart(cfsFrameDefines, m.config))
   if m.s[cfsFrameDefines].len > 0:

--- a/tests/compiler/tcppCompileToNamespace.nim
+++ b/tests/compiler/tcppCompileToNamespace.nim
@@ -1,0 +1,12 @@
+discard """
+cmd: "nim cpp --cppCompileToNamespace:foo $options -r $file"
+target: cpp
+"""
+
+# Theoretically nim could just ignore the flag cppCompileToNamespace
+# and this test would pass.  Setting ``ccodeCheck`` for a c++ target
+# doesn't work.
+
+import os
+
+echo "a" / "b"

